### PR TITLE
LPS-181556 [TR2] Automation Test | master

### PR DESCRIPTION
**Pull request by joelrocha68**
Routine Permissions
https://issues.liferay.com/browse/LPS-181555

### CODE REVIEW

**Comment by joelrocha68** [comment]
ci:test:sf

**Comment by joelrocha68** [comment]
ci:test:site-initializer-testray:nocompile

**Comment by joelrocha68** [comment]
ci:test:site-initializer-testray:nocompile

**Comment by joelrocha68** [comment]
ci:test:site-initializer-testray:nocompile

**Comment by joelrocha68** [comment]
ci:test:site-initializer-testray:nocompile

**Comment by joelrocha68** [comment]
ci:test:site-initializer-testray:nocompile

File: modules/apps/site-initializer/site-initializer-testray/site-initializer-testray-test/src/testFunctional/tests/TestrayRoutinePermissions.testcase [hunk 1]
```diff
@@ -0,0 +1,223 @@
...
+		}
+
+		var testPortalInstance = PropsUtil.get("test.portal.instance");
+
+		if (${testPortalInstance} == "true") {
+			PortalInstances.tearDownCP();
+		}
+	}
+
+	@description = "Story ID: LPS-178269 | Test Summary: Verify that the User with the 'Testray Analyst' role should not be able to create/edit/delete the routine but can view"
```
Comment (luanakarla) [inline modules/apps/site-initializer/site-initializer-testray/site-initializer-testray-test/src/testFunctional/tests/TestrayRoutinePermissions.testcase:61]: One test one intention. Could you break this test into 3 please?

File: modules/apps/site-initializer/site-initializer-testray/site-initializer-testray-test/src/testFunctional/tests/TestrayRoutinePermissions.testcase [hunk 2]
```diff
@@ -0,0 +1,223 @@
...
+		}
+
+		task ("And can not see the kebab menu") {
+			TestrayRoutines.goToRoutine(routineName = ${routineName});
+
+			AssertElementNotPresent(locator1 = "ObjectAdmin#KEBAB_MENU");
+		}
+	}
+
+	@description = "Story ID: LPS-178269 | Test Summary: Verify that the User with the 'Testray Lead' role should not be able to create/edit/delete the routine but can view"
```
Comment (luanakarla) [inline modules/apps/site-initializer/site-initializer-testray/site-initializer-testray-test/src/testFunctional/tests/TestrayRoutinePermissions.testcase:115]: Same comment as above.

File: modules/apps/site-initializer/site-initializer-testray/site-initializer-testray-test/src/testFunctional/tests/TestrayRoutinePermissions.testcase [hunk 3]
```diff
@@ -0,0 +1,223 @@
...
+			Navigator.openSiteURL(siteName = "Testray 2");
+		}
+
+		task ("And the user go to the routine page") {
+			Testray.changePagination(
+				numberItems = 20,
+				valuePagination = 150);
+
+			Testray.goToProjectPage(projectName = ${projectName});
+		}
```
Comment (luanakarla) [inline modules/apps/site-initializer/site-initializer-testray/site-initializer-testray-test/src/testFunctional/tests/TestrayRoutinePermissions.testcase:134]: This block can be placed next to the Given block. To give you more context about the use of Given, usually in this block you can add all the steps up to the test intention page.

File: modules/apps/site-initializer/site-initializer-testray/site-initializer-testray-test/src/testFunctional/tests/TestrayRoutinePermissions.testcase [hunk 4]
```diff
@@ -0,0 +1,223 @@
...
+			AssertElementNotPresent(
+				item = "Delete",
+				locator1 = "TestrayContextMenu#CONTEXT_MENU_PROJECTS_ITEMS");
+		}
+
+		task ("And can not see the kebab menu") {
+			TestrayRoutines.goToRoutine(routineName = ${routineName});
+
+			AssertElementNotPresent(locator1 = "ObjectAdmin#KEBAB_MENU");
+		}
```
Comment (luanakarla) [inline modules/apps/site-initializer/site-initializer-testray/site-initializer-testray-test/src/testFunctional/tests/TestrayRoutinePermissions.testcase:166]: One test one intention. The last block of the test should be with the `Then` task

File: modules/apps/site-initializer/site-initializer-testray/site-initializer-testray-test/src/testFunctional/tests/TestrayRoutinePermissions.testcase [hunk 5]
```diff
@@ -0,0 +1,223 @@
...
+	}
+
+	@description = "Story ID: LPS-178269 | Test Summary: Verify that the User with the 'Testray User' role should not be able to create/edit/delete the routine but can view"
+	@priority = 3
```
Comment (luanakarla) [inline modules/apps/site-initializer/site-initializer-testray/site-initializer-testray-test/src/testFunctional/tests/TestrayRoutinePermissions.testcase:170]: Same comment as above.

File: modules/apps/site-initializer/site-initializer-testray/site-initializer-testray-test/src/testFunctional/tests/TestrayRoutinePermissions.testcase [hunk 6]
```diff
@@ -0,0 +1,223 @@
...
+			Navigator.openSiteURL(siteName = "Testray 2");
+		}
+
+		task ("And the user go to the routine page") {
+			Testray.changePagination(
+				numberItems = 20,
+				valuePagination = 150);
+
+			Testray.goToProjectPage(projectName = ${projectName});
+		}
```
Comment (luanakarla) [inline modules/apps/site-initializer/site-initializer-testray/site-initializer-testray-test/src/testFunctional/tests/TestrayRoutinePermissions.testcase:188]: Same comment as above.

File: modules/apps/site-initializer/site-initializer-testray/site-initializer-testray-test/src/testFunctional/tests/TestrayRoutinePermissions.testcase [hunk 7]
```diff
@@ -0,0 +1,223 @@
...
+				item = "Delete",
+				locator1 = "TestrayContextMenu#CONTEXT_MENU_PROJECTS_ITEMS");
+		}
+
+		task ("And can not see the kebab menu") {
+			TestrayRoutines.goToRoutine(routineName = ${routineName});
+
+			AssertElementNotPresent(locator1 = "ObjectAdmin#KEBAB_MENU");
+		}
+	}
```
Comment (luanakarla) [inline modules/apps/site-initializer/site-initializer-testray/site-initializer-testray-test/src/testFunctional/tests/TestrayRoutinePermissions.testcase:221]: Same comment as above.

**Review by luanakarla — CHANGES_REQUESTED** [review]
Hey @joelrocha68 , thanks for working on it!

I requested some changes and suggested others, please can you take a look at it?

File: modules/apps/site-initializer/site-initializer-testray/site-initializer-testray-test/src/testFunctional/tests/TestrayRoutinePermissions.testcase [hunk 8]
```diff
@@ -0,0 +1,223 @@
...
+			Navigator.openSiteURL(siteName = "Testray 2");
+		}
+
+		task ("And the user go to the routine page") {
+			Testray.changePagination(
+				numberItems = 20,
+				valuePagination = 150);
+
+			Testray.goToProjectPage(projectName = ${projectName});
+		}
```
[the remark this answers is not in the document]
Comment (joelrocha68) [reply modules/apps/site-initializer/site-initializer-testray/site-initializer-testray-test/src/testFunctional/tests/TestrayRoutinePermissions.testcase]: Done

File: modules/apps/site-initializer/site-initializer-testray/site-initializer-testray-test/src/testFunctional/tests/TestrayRoutinePermissions.testcase (hunk 1 above)
Comment (joelrocha68) [reply modules/apps/site-initializer/site-initializer-testray/site-initializer-testray-test/src/testFunctional/tests/TestrayRoutinePermissions.testcase]: Done

File: modules/apps/site-initializer/site-initializer-testray/site-initializer-testray-test/src/testFunctional/tests/TestrayRoutinePermissions.testcase (hunk 2 above)
Comment (joelrocha68) [reply modules/apps/site-initializer/site-initializer-testray/site-initializer-testray-test/src/testFunctional/tests/TestrayRoutinePermissions.testcase]: Done

File: modules/apps/site-initializer/site-initializer-testray/site-initializer-testray-test/src/testFunctional/tests/TestrayRoutinePermissions.testcase (hunk 3 above)
Comment (joelrocha68) [reply modules/apps/site-initializer/site-initializer-testray/site-initializer-testray-test/src/testFunctional/tests/TestrayRoutinePermissions.testcase]: Done

File: modules/apps/site-initializer/site-initializer-testray/site-initializer-testray-test/src/testFunctional/tests/TestrayRoutinePermissions.testcase (hunk 4 above)
Comment (joelrocha68) [reply modules/apps/site-initializer/site-initializer-testray/site-initializer-testray-test/src/testFunctional/tests/TestrayRoutinePermissions.testcase]: Done

File: modules/apps/site-initializer/site-initializer-testray/site-initializer-testray-test/src/testFunctional/tests/TestrayRoutinePermissions.testcase (hunk 5 above)
Comment (joelrocha68) [reply modules/apps/site-initializer/site-initializer-testray/site-initializer-testray-test/src/testFunctional/tests/TestrayRoutinePermissions.testcase]: Done

File: modules/apps/site-initializer/site-initializer-testray/site-initializer-testray-test/src/testFunctional/tests/TestrayRoutinePermissions.testcase (hunk 6 above)
Comment (joelrocha68) [reply modules/apps/site-initializer/site-initializer-testray/site-initializer-testray-test/src/testFunctional/tests/TestrayRoutinePermissions.testcase]: Done

File: modules/apps/site-initializer/site-initializer-testray/site-initializer-testray-test/src/testFunctional/tests/TestrayRoutinePermissions.testcase (hunk 7 above)
Comment (joelrocha68) [reply modules/apps/site-initializer/site-initializer-testray/site-initializer-testray-test/src/testFunctional/tests/TestrayRoutinePermissions.testcase]: Done

**Comment by joelrocha68** [comment]
ci:test:sf

**Comment by joelrocha68** [comment]
ci:test:sf

**Review by luanakarla — APPROVED** [review]

**Review by luanakarla — APPROVED** [review]
Hey @joelrocha68 , thanks for working on it!

The code looks good to me!

I'm forwarding it.

**Comment by joelrocha68** [comment]
ci:forward

**Comment by joelrocha68** [comment]
ci:forward

**Comment by joelrocha68** [comment]
ci:stop

**Comment by joelrocha68** [comment]
ci:close

**Comment by joelrocha68** [comment]
ci:forward

**Comment by joelrocha68** [comment]
ci:forward

**Pull request closed without merging**